web/profiles/joinup/src/Plugin/Block/FrontpageBlock.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Drupal\joinup\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\search_api\Entity\Index;
+use Drupal\search_api\Query\ResultSetInterface;
+use Drupal\search_api\SearchApiException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Session\AccountProxy;
+use Drupal\og\MembershipManager;
+
+/**
+ * Provides a 'FrontpageBlock' block.
+ *
+ * @Block(
+ *  id = "frontpage_block",
+ *  admin_label = @Translation("Frontpage block"),
+ * )
+ */
+class FrontpageBlock extends BlockBase implements ContainerFactoryPluginInterface {
```

Comment:
I would use a more explicative name for this block. The scope of this block is to show recommended content to the user, and it doesn't have any technical limitation regarding being in the homepage. So, I would call it something like `RecommendedContentBlock`.